x/oracle/abci_test.go

```diff
@@ -0,0 +1,109 @@
+package oracle_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+	"github.com/cosmos/cosmos-sdk/x/staking"
+	"github.com/cosmos/cosmos-sdk/x/staking/teststaking"
+	"github.com/stretchr/testify/suite"
+	"github.com/tendermint/tendermint/crypto/secp256k1"
+	tmrand "github.com/tendermint/tendermint/libs/rand"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	umeeapp "github.com/umee-network/umee/v3/app"
+	appparams "github.com/umee-network/umee/v3/app/params"
+	"github.com/umee-network/umee/v3/x/oracle"
+	"github.com/umee-network/umee/v3/x/oracle/types"
+)
+
+const (
+	displayDenom string = appparams.DisplayDenom
+	bondDenom    string = appparams.BondDenom
+)
+
+type IntegrationTestSuite struct {
+	suite.Suite
+
+	ctx sdk.Context
+	app *umeeapp.UmeeApp
+}
+
+const (
+	initialPower = int64(10000000000)
+)
+
+func (s *IntegrationTestSuite) SetupTest() {
+	require := s.Require()
+	isCheckTx := false
+	app := umeeapp.Setup(s.T(), isCheckTx, 1)
+	ctx := app.BaseApp.NewContext(isCheckTx, tmproto.Header{
+		ChainID: fmt.Sprintf("test-chain-%s", tmrand.Str(4)),
+		Height:  int64(types.DefaultMedianPeriod) - 1,
+	})
+
+	oracle.InitGenesis(ctx, app.OracleKeeper, *types.DefaultGenesisState())
+
+	sh := teststaking.NewHelper(s.T(), ctx, *app.StakingKeeper)
+	sh.Denom = bondDenom
+	amt := sdk.TokensFromConsensusPower(100, sdk.DefaultPowerReduction)
+
+	// mint and send coins to validators
+	require.NoError(app.BankKeeper.MintCoins(ctx, minttypes.ModuleName, initCoins))
+	require.NoError(app.BankKeeper.SendCoinsFromModuleToAccount(ctx, minttypes.ModuleName, addr, initCoins))
+	require.NoError(app.BankKeeper.MintCoins(ctx, minttypes.ModuleName, initCoins))
+	require.NoError(app.BankKeeper.SendCoinsFromModuleToAccount(ctx, minttypes.ModuleName, addr2, initCoins))
+
+	sh.CreateValidator(valAddr, valPubKey, amt, true)
+	sh.CreateValidator(valAddr2, valPubKey2, amt, true)
+
+	staking.EndBlocker(ctx, *app.StakingKeeper)
+
+	s.app = app
+	s.ctx = ctx
+}
+
+// Test addresses
+var (
+	valPubKeys = simapp.CreateTestPubKeys(2)
+
+	valPubKey = valPubKeys[0]
+	pubKey    = secp256k1.GenPrivKey().PubKey()
+	addr      = sdk.AccAddress(pubKey.Address())
+	valAddr   = sdk.ValAddress(pubKey.Address())
+
+	valPubKey2 = valPubKeys[1]
+	pubKey2    = secp256k1.GenPrivKey().PubKey()
+	addr2      = sdk.AccAddress(pubKey2.Address())
+	valAddr2   = sdk.ValAddress(pubKey2.Address())
+
+	initTokens = sdk.TokensFromConsensusPower(initialPower, sdk.DefaultPowerReduction)
+	initCoins  = sdk.NewCoins(sdk.NewCoin(bondDenom, initTokens))
+)
```

Comment:
let's reduce copy paste. We can export common things to  some utility functions and create a structure with data for a 2-node network.

Reply:
I'll do this cleanup in the next pr where I'll be adding more tests for the endblocker